### PR TITLE
Issue 154 model has contract and model has constraints

### PR DESCRIPTION
**Pull request by xasm83**
- Added contract enforced check and constraints check(type and columns fields). Copy pasted the structure from other checks
- Added corresponding docs.
- JSON parsing for parameters as there is no other way to define a complex parameter for constraints.
- New unit tests + manual tests on jaffle_shop

### CODE REVIEW

**Comment by xasm83** [comment]
@BAntonellini  Could you please check it or tag someone who could do that. Thanks!

File: dbt_checkpoint/check_model_has_contract.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    status_code = 0
+    sqls = get_model_sqls(paths, manifest, include_disabled)
+    filenames = set(sqls.keys())
+
+    # get manifest nodes that pre-commit found as changed
+    models = get_models(manifest, filenames, include_disabled=include_disabled)
+
+    for model in models:
+        config = model.node.get("config", {})
+        if not config.get('contract') or not config.get('contract').get('enforced'):
```
Comment (BAntonellini) [inline dbt_checkpoint/check_model_has_contract.py]: couldn't this be simplified to the following?

```python
if not config.get('contract', {}).get('enforced')
```

File: dbt_checkpoint/check_model_has_constraints.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        if key not in model_constraint or model_constraint[key] != value:
+            return False
+    return True
+
+
+def has_constraints(constraints:Sequence[Dict[str, Any]], model:Model) -> bool:
+    model_constraints = model.node.get("constraints", [])
+    for constraint in constraints:
+        found = False
+        for model_consraint in model_constraints:
```
Comment (BAntonellini) [inline dbt_checkpoint/check_model_has_constraints.py]: `model_consraint` seems to have a typo (no `t`)

**Review by BAntonellini — CHANGES_REQUESTED** [review]
Nice contribution @xasm83, thank you for it.

I've found 2 minor, non-breaking, things. Let's discuss before merging

File: dbt_checkpoint/check_model_has_contract.py (hunk 1 above)
Comment (xasm83) [reply dbt_checkpoint/check_model_has_contract.py]: sure, adjusted

File: dbt_checkpoint/check_model_has_constraints.py (hunk 2 above)
Comment (xasm83) [reply dbt_checkpoint/check_model_has_constraints.py]: fixed

**Comment by xasm83** [comment]
@BAntonellini  thanks for reviewing! Added the requested changes

**Review by BAntonellini — APPROVED** [review]

**Pull request merged**